include/node.hpp

```diff
@@ -598,7 +605,7 @@ struct node : protected std::tuple<Arguments...> {
 
     constexpr void
     forward_tags() noexcept {
-        if (!_output_tags_changed) {
+        if (!_output_tags_changed && !_input_tags_present) {
```

Comment:
Wasn't this (and some other parts) a part of 166?

Reply:
Yes, this is the third PR in a row. The built upon each other but are functionally not directly related.